suite-native/receive/src/components/ReceiveAddressCard/useReceiveAddressCardAnimations.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect } from 'react';
+import { withDelay, withTiming, useAnimatedStyle, useSharedValue } from 'react-native-reanimated';
+
+import { QRCODE_SIZE } from '@suite-native/qr-code/src';
+import { useNativeStyles } from '@trezor/styles';
+
+import { ReceiveProgressStep } from '../../hooks/useReceiveProgressSteps';
+
+const DEVICE_MIN_PADING_TOP = 0;
+const DEVICE_MID_PADING_TOP = 20;
+const DEVICE_MAX_PADING_TOP = 36;
+
+export const useReceiveAddressCardAnimations = ({
+    receiveProgressStep,
+}: {
+    receiveProgressStep: ReceiveProgressStep;
+}) => {
+    const { utils } = useNativeStyles();
+    const qrHeight = useSharedValue(0);
+    const qrOpacity = useSharedValue(0);
+    const buttonsOpacity = useSharedValue(0);
+    const addressPaddingTop = useSharedValue(DEVICE_MIN_PADING_TOP);
+    const isShowingQRCode = useSharedValue(false);
+    const isShowedDeviceQRStep = receiveProgressStep === ReceiveProgressStep.ApprovedOnTrezor;
+    const isShowedPortfolioQRStep =
+        receiveProgressStep === ReceiveProgressStep.ShownPortfolioAddress;
+
+    useEffect(() => {
+        let timer: ReturnType<typeof setTimeout>;
+
+        if (isShowedDeviceQRStep) {
+            qrHeight.value = QRCODE_SIZE;
+            qrOpacity.value = 1;
+            buttonsOpacity.value = 1;
+            addressPaddingTop.value = DEVICE_MIN_PADING_TOP;
+            timer = setTimeout(
```

Comment:
What is timeout good for here?

Reply:
to set isShowingQRCode

partially updated in 9fe8f1e341